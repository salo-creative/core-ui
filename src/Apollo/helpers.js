import { get, hasIn, forEach, isEmpty, find } from 'lodash';

// HELPERS & CONSTANTS
export const isLoading = ({ networkStatus, skip = false }) => !skip && (
  // Initial load
  (networkStatus === 1)
  // Reload due to variables change
  || (networkStatus === 2)
  // Reload due to refetch
  || (networkStatus === 4)
);

export const isError = ({ networkStatus, error, skip = false }) => (
  error || (!skip && (networkStatus === 8)) // Network error (exclude skip as that also causes status of 8)
);

export const getRequestStatus = ({ networkStatus, error, skip = false }) => {
  return {
    requestLoading: isLoading({ networkStatus, skip }),
    requestError: isError({ networkStatus, error, skip })
  };
};

export const parseApolloError = ({ error, propName = 'APOLLO_ERROR' }) => {
  let code = 400;
  let message = 'Something went wrong!';
  const errors = [];
  const extra = {};
  let name = propName; // For custom sentry logging
  if (error.graphQLErrors) {
    // Find custom errors if they exist
    const customError = find(error.graphQLErrors, { message: 'SC_ERROR' });

    if (hasIn(customError, 'path')) {
      extra.path = customError.path.join(' -> ');
      name = propName === 'APOLLO_ERROR' ? `APOLLO_ERROR: ${ customError.path[0] }` : propName;
    }

    code = get(customError, 'extensions.status', code);

    // Handle custom error message
    const serverMessage = get(customError, 'extensions.message');
    if (serverMessage && typeof serverMessage === 'string') {
      message = serverMessage;
    } else if (typeof serverMessage === 'object') {
      message = serverMessage.message; // eslint-disable-line
      code = serverMessage.name === 'ValidationError' ? 422 : 400;
    }

    // Handle custom errors array
    const serverErrorsMessages = get(customError, 'extensions.errors');
    if (!isEmpty(serverErrorsMessages)) {
      forEach(serverErrorsMessages, errMessage => {
        if (typeof errMessage === 'string') errors.push(errMessage);
      });
    }
  }

  return {
    code,
    errors,
    extra,
    message,
    name
  };
};