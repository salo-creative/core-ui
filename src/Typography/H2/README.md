# Heading two

Heading one is a pure styled component render. Renders a standard `<h2>` with themed styling

---

## Usage

Install

```javascript
yarn add @salo/core-ui
```

Include

```javascript
import H2 from '@salo/core-ui/Typography/h2';
```

Implement as follows

```javascript
<H2>This is a title</H2>
```

### For full prop types and usage see storybook info/knobs