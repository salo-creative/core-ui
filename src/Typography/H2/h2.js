import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import FluidType from '../../helpers/fluidType';

const StyledH2 = styled.h2`
  width: 100%;
  color: ${ ({ color, theme }) => color || theme.black };
  text-align: ${ ({ align }) => align };
  margin: ${ ({ margin }) => margin };
  padding: ${ ({ padding }) => padding };

  ${ ({ fontSize, lineHeight, minFont, maxFont, minLine, maxLine }) => {
    if (fontSize === '' && lineHeight === '') {
      return `${ FluidType({ fontSize: [minFont, maxFont], lineHeight: [minLine, maxLine] }) }`;
    }
    return `
      font-size: ${ fontSize };
      line-height: ${ lineHeight };
    `;
  } }
`;

const H2 = ({ children, ...props }) => {
  return <StyledH2 { ...props }>{ children }</StyledH2>;
};

H2.propTypes = {
  color: PropTypes.string,
  align: PropTypes.string,
  fontSize: PropTypes.string,
  lineHeight: PropTypes.string,
  margin: PropTypes.string,
  padding: PropTypes.string,
  minFont: PropTypes.number,
  minLine: PropTypes.number,
  maxFont: PropTypes.number,
  maxLine: PropTypes.number
};

H2.defaultProps = {
  color: '',
  align: 'left',
  fontSize: '',
  lineHeight: '',
  margin: '30px 0 10px',
  padding: '0',
  minFont: 27,
  maxFont: 40,
  minLine: 31,
  maxLine: 47
};

H2.displayName = 'H2';

export default H2;