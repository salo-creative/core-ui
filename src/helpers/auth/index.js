import { isEmpty, get } from 'lodash';
import Cookies from 'universal-cookie';

const cookies = new Cookies();

/**
 * GET USER COOKIES
 * get the session cookies from the cookies available
 */
export const getSessionCookies = () => {
  let tokens = null;
  const jwt = cookies.get('SCSession');
  if (!isEmpty(jwt)) {
    tokens = { jwt };
  }
  return tokens;
};

/**
 * TOKEN EXPIRED
 * Check if the jwt has expired (expects a time string NOT THE JWT)
 */
export const tokenExpired = (expiry) => {
  if (!expiry) return false;
  const timeDiff = Math.round((Date.now() - expiry) / 1000);
  return timeDiff >= 7 * 24 * 60 * 60;
};

/**
 * GET JWT
 * Get the jwt token for the local cookies
 */
export const getJwt = () => {
  const jwt = cookies.get('SCSession');
  if (jwt && jwt.ts && !tokenExpired(jwt.ts)) {
    return get(jwt, 't', '');
  }
  // No cookie found or malformed cookie.
  return '';
};