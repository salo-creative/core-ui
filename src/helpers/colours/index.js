import { darken } from 'polished';

/**
 * COLOUR DEFINITIONS
 *
 * Fixed colour definitions to ensure
 * centralised definition of consts
 */
export const colours = {
  blue: '#00ACED',
  brightBlue: '#00D2FF',
  charcoal: '#333333',
  darkGrey: '#6c757d',
  green: '#31D238',
  grey: '#e9ecef',
  orange: '#f47a3e',
  paleGrey: '#fafcff',
  pink: '#EF358A',
  purple: '#62358A',
  red: '#f15b60',
  yellow: '#FFAE00'
};

export const saloTheme = {
  primary: colours.blue,
  secondary: colours.pink,
  success: colours.green,
  error: colours.red,
  warning: colours.yellow,
  info: colours.grey,
  font: colours.charcoal,
  fontInverse: '#fff',
  bodyFont: '\'Brother\', sans-serif',
  headerFont: '\'Brother\', sans-serif',
  ...colours
};

/**
 * BOX SHADOW
 *
 * Centralised definition for shadows
 * @param {*} size - string value for the size. [default | medium, | large]
 * @returns box shadow css property string
 */
export const boxShadow = (size) => {
  switch (size) {
    case 'large':
    case 'small':
    case 'default':
    default:
      return '2px 2px 2px 0px rgba(0,0,0,0.2)';
  }
};

/**
 * BUTTON THEMES
 *
 * Definition for the button themes from the predefined colours
 */
export const buttonThemes = ({ primary, secondary, font, fontInverse, success, error }) => ({
  white: {
    border: '#fff',
    background: '#fff',
    color: font,
    borderHov: colours.paleGrey,
    backgroundHov: colours.paleGrey,
    colorHov: font,
    shadow: true
  },
  clear: {
    border: font,
    background: 'transparent',
    color: font,
    borderHov: font,
    backgroundHov: 'rgba(0,0,0,0.1)',
    colorHov: font,
    shadow: false
  },
  clear_inverse: {
    border: fontInverse,
    background: 'transparent',
    color: fontInverse,
    borderHov: darken(0.3, fontInverse),
    backgroundHov: 'transparent',
    colorHov: darken(0.3, fontInverse),
    shadow: false
  },
  text: {
    border: 'transparent',
    background: 'transparent',
    color: colours.darkGrey,
    borderHov: 'transparent',
    backgroundHov: 'transparent',
    colorHov: font,
    shadow: false
  },
  text_inverse: {
    border: 'transparent',
    background: 'transparent',
    color: colours.paleGrey,
    borderHov: 'transparent',
    backgroundHov: 'transparent',
    colorHov: fontInverse,
    shadow: false
  },
  primary: {
    border: primary,
    background: primary,
    color: fontInverse,
    borderHov: darken(0.15, primary),
    backgroundHov: darken(0.15, primary),
    colorHov: fontInverse,
    shadow: true
  },
  secondary: {
    border: secondary,
    background: secondary,
    color: fontInverse,
    borderHov: darken(0.15, secondary),
    backgroundHov: darken(0.15, secondary),
    colorHov: fontInverse,
    shadow: true
  },
  error: {
    border: error,
    background: error,
    color: fontInverse,
    borderHov: darken(0.15, error),
    backgroundHov: darken(0.15, error),
    colorHov: fontInverse,
    shadow: true
  },
  success: {
    border: success,
    background: success,
    color: fontInverse,
    borderHov: darken(0.15, success),
    backgroundHov: darken(0.15, success),
    colorHov: fontInverse,
    shadow: true
  }
});