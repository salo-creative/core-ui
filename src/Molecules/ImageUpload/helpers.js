import { dataURItoBlob } from '../../helpers/files';

export const onImageDrop = async ({
  files,
  maxFileSize,
  onUpload,
  rejected,
  setFileSizeError
}) => {
  if (rejected.length > 0) {
    if (rejected[0].size >= maxFileSize) {
      setFileSizeError(true);
    }
    return;
  }
  
  if (!files.length) return;
  
  setFileSizeError(false);
  
  const file = files[0];
  const img = new Image();
  
  img.onload = (e) => {
    const sizes = {
      width: e.target.width,
      height: e.target.height
    };
  
    const reader = new FileReader();
  
    reader.onload = (event) => {
      // Image service requires a binary, not base64 so we need to convert back
      const fileBlob = dataURItoBlob(event.target.result);
      onUpload({
        blob: fileBlob, base64: event.target.result, sizes
      });
    };
    reader.readAsDataURL(file);
  };
  
  img.src = URL.createObjectURL(file);
};