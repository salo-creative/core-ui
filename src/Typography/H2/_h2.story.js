import React from 'react';
import { storiesOf } from '@storybook/react';

// load tests
import { withTests } from '@storybook/addon-jest';
import { withKnobs, text, select, color } from '@storybook/addon-knobs';
import results from '../../../.storybook/jest-test-results.json';

// FEATURED COMPONENT
import { H2 } from '../../index';

// README //
import README from './README.md';

// Start of story logic
const stories = storiesOf('Atoms | Typography/H2', module);
stories.addDecorator(withKnobs);
stories.addDecorator(withTests({ results }));
stories.addParameters({ jest: ['h2'] });

stories.add(
  'Basic',
  (() => {
    const align = select('Align', {
      'Left': 'left',
      'Right': 'right',
      'Center': 'center'
    }, 'center');
    const fontSize = text('Font size', '');
    const colour = color('Colour', '#262729');
    const margin = text('Margin', '30px 0 10px');
    return (
      <H2
        color={ colour }
        fontSize={ fontSize }
        align={ align }
        margin={ margin }
      >
        H2 Headline <br /> With a second line
      </H2>
    );
  }), { notes: README }
);