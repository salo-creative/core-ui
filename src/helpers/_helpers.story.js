import breakpoints from './breakpoints/_breakpoints.story';
import colours from './colours/_colours.story';
import environments from './environments/_environments.story';