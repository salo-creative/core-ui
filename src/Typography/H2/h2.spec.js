import React from 'react';
import 'jest-styled-components';

// HELPERS
import { fluidTypeVals, renderWithTheme } from '../../../test';

// COMPONENT
import H2 from './index';
import { colours } from '../../helpers/colours';

test('Render default H2 tag and styles', async () => {
  // Variables
  const minFS = 27;
  const maxFS = 40;
  const minLH = 31;
  const maxLH = 47;
  // Render
  const { getByTestId, container } = renderWithTheme(
    <H2 data-testid='test-text'>A tasty treat</H2>
  );
  const el = getByTestId('test-text');
  const fontSizes = fluidTypeVals({ minLH, maxLH, maxFS, minFS });
  // Assert
  expect(el).toHaveTextContent('A tasty treat');
  expect(el).toHaveStyleRule('color', colours.black);
  expect(el).toHaveStyleRule('padding', '0');
  expect(el).toHaveStyleRule('margin', '30px 0 10px');
  expect(el).toHaveStyleRule('font-size', `${ minFS }px`);
  expect(el).toHaveStyleRule('font-size', fontSizes.minFont, { media: 'screen and (min-width:400px)' });
  expect(el).toHaveStyleRule('font-size', `${ maxFS }px`, { media: 'screen and (min-width:1200px)' });
  expect(el).toHaveStyleRule('line-height', `${ minLH }px`);
  expect(el).toHaveStyleRule('line-height', fontSizes.minLineHeight, { media: 'screen and (min-width:400px)' });
  expect(el).toHaveStyleRule('line-height', `${ maxLH }px`, { media: 'screen and (min-width:1200px)' });
  expect(container.firstChild).toMatchSnapshot();
});

test('Render H2 tag with core props', async () => {
  // Render
  const { getByTestId, container } = renderWithTheme(
    <H2
      data-testid='test-text'
      fontSize='30px'
      color='#333'
      lineHeight='40px'
      align='right'
      margin='10px 0 10px'
      padding='10px 20px'
    >
      A tasty treat
    </H2>
  );
  const el = getByTestId('test-text');
  // Assert
  expect(el).toHaveStyleRule('padding', '10px 20px');
  expect(el).toHaveStyleRule('margin', '10px 0 10px');
  expect(el).toHaveStyleRule('font-size', '30px');
  expect(el).toHaveStyleRule('line-height', '40px');
  expect(el).toHaveStyleRule('text-align', 'right');
  expect(el).toHaveStyleRule('color', '#333');
  expect(container.firstChild).toMatchSnapshot();
});

test('Render H2 tag with fluid type props', async () => {
  // Variables
  const minFS = 60;
  const maxFS = 100;
  const minLH = 70;
  const maxLH = 110;
  // Render
  const { getByTestId, container } = renderWithTheme(
    <H2
      data-testid='test-text'
      minFont={ minFS }
      maxFont={ maxFS }
      minLine={ minLH }
      maxLine={ maxLH }
    >
      A tasty treat
    </H2>
  );
  const el = getByTestId('test-text');
  // Assert
  const fontSizes = fluidTypeVals({ minLH, maxLH, maxFS, minFS });
  expect(el).toHaveStyleRule('font-size', `${ minFS }px`);
  expect(el).toHaveStyleRule('font-size', fontSizes.minFont, { media: 'screen and (min-width:400px)' });
  expect(el).toHaveStyleRule('font-size', `${ maxFS }px`, { media: 'screen and (min-width:1200px)' });
  expect(el).toHaveStyleRule('line-height', `${ minLH }px`);
  expect(el).toHaveStyleRule('line-height', fontSizes.minLineHeight, { media: 'screen and (min-width:400px)' });
  expect(el).toHaveStyleRule('line-height', `${ maxLH }px`, { media: 'screen and (min-width:1200px)' });
  expect(container.firstChild).toMatchSnapshot();
});