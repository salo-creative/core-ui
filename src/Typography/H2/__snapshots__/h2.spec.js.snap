// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render H2 tag with core props 1`] = `
.c0 {
  width: 100%;
  color: #333;
  text-align: right;
  margin: 10px 0 10px;
  padding: 10px 20px;
  font-weight: 700;
  font-size: 30px;
  line-height: 40px;
}

<h2
  class="c0"
  color="#333"
  data-testid="test-text"
  font-size="30px"
>
  A tasty treat
</h2>
`;

exports[`Render H2 tag with fluid type props 1`] = `
.c0 {
  width: 100%;
  color: #333333;
  text-align: left;
  margin: 30px 0 10px;
  padding: 0;
  font-weight: 700;
  font-size: 60px;
  line-height: 70px;
}

@media screen and (min-width:400px) {
  .c0 {
    font-size: calc(60px + (100 - 60) * (100vw - 400px) / (1200 - 400));
    line-height: calc(70px + (110 - 70) * (100vw - 400px) / (1200 - 400));
  }
}

@media screen and (min-width:1200px) {
  .c0 {
    font-size: 100px;
    line-height: 110px;
  }
}

<h2
  class="c0"
  color=""
  data-testid="test-text"
  font-size=""
>
  A tasty treat
</h2>
`;

exports[`Render default H2 tag and styles 1`] = `
.c0 {
  width: 100%;
  color: #333333;
  text-align: left;
  margin: 30px 0 10px;
  padding: 0;
  font-weight: 700;
  font-size: 27px;
  line-height: 31px;
}

@media screen and (min-width:400px) {
  .c0 {
    font-size: calc(27px + (40 - 27) * (100vw - 400px) / (1200 - 400));
    line-height: calc(31px + (47 - 31) * (100vw - 400px) / (1200 - 400));
  }
}

@media screen and (min-width:1200px) {
  .c0 {
    font-size: 40px;
    line-height: 47px;
  }
}

<h2
  class="c0"
  color=""
  data-testid="test-text"
  font-size=""
>
  A tasty treat
</h2>
`;
